### PR TITLE
Make leader-election flag consistent with other sidecar

### LINKED ISSUE #401: Arg "--enable-leader-election" is not consistent with other sidecars
All the other sidecars use "--leader-election".

### PR DESCRIPTION
**Pull request by Madhu-1**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR removes the `--enable-leader-election`  and `leader-election-type` flag and add `leader-election` flag as other sidecar containers.

**Which issue(s) this PR fixes**:

Fixes #401

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
This introduces a new flag `leader-election` which is the same as in other sidecar containers and also removes the `enable-leader-election` and `leader-election-type` flag
```

### CODE REVIEW

File: cmd/csi-provisioner/csi-provisioner.go [hunk 1]
```diff
@@ -64,7 +64,10 @@ var (
 	operationTimeout     = flag.Duration("timeout", 10*time.Second, "Timeout for waiting for creation or deletion of a volume")
 	_                    = deprecatedflags.Add("provisioner")
 
-	enableLeaderElection    = flag.Bool("enable-leader-election", false, "Enables leader election. If leader election is enabled, additional RBAC rules are required. Please refer to the Kubernetes CSI documentation for instructions on setting up these RBAC rules.")
+	_ = deprecatedflags.Add("enable-leader-election")
+
+	enableLeaderElection = flag.Bool("leader-election", false, "Enables leader election. If leader election is enabled, additional RBAC rules are required. Please refer to the Kubernetes CSI documentation for instructions on setting up these RBAC rules.")
```
Comment (jsafrane) [inline cmd/csi-provisioner/csi-provisioner.go]: This is a breaking change that requires bump to v2.0.
@msau42, can we expect that the next release will be 2.0?

@Madhu-1, depending on the answer to above question, would it be possible to support both `enable-leader-election` and `leader-election` at the same time until we're ready for v2.0? It's OK to throw error if both are specified.

Comment (Madhu-1) [reply cmd/csi-provisioner/csi-provisioner.go]: @jsafrane yes I can do that, will wait for @msau42 response

Comment (msau42) [reply cmd/csi-provisioner/csi-provisioner.go:69]: I just released a 1.6, and mentioned this in the deprecation notes. We can proceed and just break it.

File: cmd/csi-provisioner/csi-provisioner.go [hunk 2]
```diff
@@ -64,7 +64,10 @@ var (
 	operationTimeout     = flag.Duration("timeout", 10*time.Second, "Timeout for waiting for creation or deletion of a volume")
 	_                    = deprecatedflags.Add("provisioner")
 
-	enableLeaderElection    = flag.Bool("enable-leader-election", false, "Enables leader election. If leader election is enabled, additional RBAC rules are required. Please refer to the Kubernetes CSI documentation for instructions on setting up these RBAC rules.")
+	_ = deprecatedflags.Add("enable-leader-election")
```
Comment (msau42) [inline cmd/csi-provisioner/csi-provisioner.go:67]: I think we can just remove the old arg.

File: cmd/csi-provisioner/csi-provisioner.go [hunk 3]
```diff
@@ -64,7 +64,10 @@ var (
...
+
+	enableLeaderElection = flag.Bool("leader-election", false, "Enables leader election. If leader election is enabled, additional RBAC rules are required. Please refer to the Kubernetes CSI documentation for instructions on setting up these RBAC rules.")
+
 	leaderElectionType      = flag.String("leader-election-type", "endpoints", "the type of leader election, options are 'endpoints' (default) or 'leases' (strongly recommended). The 'endpoints' option is deprecated in favor of 'leases'.")
```
Comment (msau42) [inline cmd/csi-provisioner/csi-provisioner.go:71]: And remove the "type" arg. We will only support leases going forward.

**Comment by Madhu-1** [comment]
@msau42 changes done PTAL

**Comment by msau42** [comment]
@Madhu-1 seems like the travis pr job is stuck. Can you try pushing a change to see if that helps the job restart?

**Comment by msau42** [comment]
Travis pull jobs have been disabled on this repo now. Force merging.

**Pull request merged**